### PR TITLE
refactor: mixin for page metadata

### PR DESCRIPTION
**Pull request by rwd**
goals:
1. get rid of the clunky head functions in page Vues, and just have one in the layout
2. have a mechanism whereby components like the error message one can influence the page title

refactor:
* introduces a new mixin, `pageMeta`, which constructs the contents of `head` based on a computed property `pageMeta`, and checks for a fetchState error for the page title
* in every page Vue:
  * import and use the `pageMeta` mixin
  * add a computed property `pageMeta` which is an object of metadata for use in the HTML head
  * remove its `head` function
* remove the `pageHeadTitle` plugin and let the mixin combine the page title and site name

### CODE REVIEW

**Comment by rwd** [comment]
Re EC-6031

**Review by LeoniePeters — CHANGES_REQUESTED** [review]

File: packages/portal/src/pages/collections/_type/_.vue [hunk 1]
```diff
@@ -174,35 +176,28 @@
...
+      pageMeta() {
+        return {
+          title: this.title.values?.[0],
+          description: this.descriptionText,
+          ogType: 'article'
+        };
+      },
+      _page() {
+        return this.$page;
+      },
```
[the remark this answers is not in the document]
Comment (rwd) [reply packages/portal/src/pages/collections/_type/_.vue]: Nothing; removed.

File: packages/portal/tests/unit/components/generic/ErrorMessage.spec.js [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
 
 const localVue = createLocalVue();
 localVue.use(BootstrapVue);
 
-const factory = (propsData = {}) => shallowMount(ErrorMessage, {
+const factory = (propsData = {}) => shallowMountNuxt(ErrorMessage, {
   localVue,
   propsData,
   mocks: {
+    $store: {
```
Comment (LeoniePeters) [inline packages/portal/tests/unit/components/generic/ErrorMessage.spec.js:14]: Clean up the `$store` mock together with `sinon` and `shallowMountNuxt`

File: packages/portal/tests/unit/layouts/default.spec.js [hunk 3]
```diff
@@ -29,6 +31,9 @@ const factory = (options = {}) => shallowMountNuxt(layout, {
   mocks: {
     $store: {
       state: {
+        pageMeta: {
+          data: {}
+        },
```
Comment (LeoniePeters) [inline packages/portal/tests/unit/layouts/default.spec.js:36]: Revert

File: packages/portal/tests/unit/layouts/error.spec.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      expect(text.includes(error.message)).toBe(true);
+    });
+  });
+
+  describe('head', () => {
+    describe('title', () => {
+      it('defaults to "Error" (translated)', () => {
+        const pageMeta = {};
+        const wrapper = factory({ propsData: { error } });
+        wrapper.vm.$store.state.pageMeta.data = pageMeta;
```
Comment (LeoniePeters) [inline packages/portal/tests/unit/layouts/error.spec.js:51]: Remove this line

**Review by LeoniePeters — CHANGES_REQUESTED** [review]

**Review by LeoniePeters — APPROVED** [review]

**Pull request merged**